Quadriga/src/main/java/edu/asu/spring/quadriga/service/network/impl/NetworkTransformationManager.java

```diff
@@ -299,29 +300,37 @@ private ITransformedNetwork getFinalTransformedNetwork(ITransformedNetwork trans
         // Add all the statement id of the selected node to a set.
         // The alternative id list if not empty, is sure to contain the concept
         // id corresponding to the searched concept.
-        Set<String> statementIdSearchSet = new HashSet<String>();
+        List<Set<String>> statementIdSearchSetList = new ArrayList<Set<String>>();
+        Set<String> statementIdSet = new HashSet<String>();
         List<Node> searchedNodes = new ArrayList<Node>();
         for (Node node : transformedNetwork.getNodes().values()) {
             // Check if the node's concept id is present in the alternative id
             // list of the concept.
+            Set<String> statementIdSearchSet = new HashSet<String>();
             for (String alternativeId : alternativeIdsForConcept) {
                 if (alternativeId.equals(node.getConceptId())) {
                     searchedNodes.add(node);
                     statementIdSearchSet.addAll(node.getStatementIds());
+                    statementIdSearchSetList.add(statementIdSearchSet);
                     break;
                 }
             }
         }
-
+        if(statementIdSearchSetList.size() >= 1){
+            statementIdSet = statementIdSearchSetList.get(0);
+            for(int i = 1; i < statementIdSearchSetList.size() ; i++){
+                statementIdSet = Sets.intersection(statementIdSet, statementIdSearchSetList.get(i));
+            }
+        } 
```

Comment:
I think this is wrong. You treat alternative ids the ids of another concept equally. What if a concept appears twice in a statement? Wouldn't that result in the statement being included although it doesn't contain both concepts? I don't think you need to intersect anything, just pass a list of lists or sets into the method and then make sure to only keep the statement ids that you've seen for the previous concepts.